### DISCUSSION TITLE
registerBotCommands not working

### DISCUSSION THREAD

**Discussion by mxviz2** [def-studio/telegraph#107] [Q&A] [answered]
I am trying to set menu with commands

```php
class TestWebhook extends \DefStudio\Telegraph\Handlers\WebhookHandler
{
    public function start()
    {
        $this->bot->registerCommands([
            'info' => 'Info',
            'registration' => 'Registration',
            'search' => 'Search'
        ]);
    }
}
```

and nothing changes

**Comment by fabio-ivona** [comment, marked as answer]
`registerCommands` is use to register bot commands in the same way you would do with botfather, it shouldn't be used inside a webhook handler

what's the result you are trying to achieve?

ps: I'm moving this into discussions, issues are reserved for bugs

**Reply by mxviz2** [reply]
I thought I could set different menus depending on the user's role. Not in this piece of code, but in the future. And I think it's not possible, right?

**Reply by fabio-ivona** [reply]
not with current telegram bot APIs, it's not this package lack of support, simply it is not provided by telegram, sorry

**Reply by mxviz2** [reply]
Yes, I got it, thank you
